src/components/SwapWidget/MarketOrder/index.tsx

```diff
@@ -163,6 +172,32 @@ const MarketOrder: React.FC<Props> = ({
     [onUserInput],
   );
 
+  // setting default tokens
+  const filterCurrencies = (symbolName: string): Currency[] => {
+    const tokens = Object.values(allTokens);
+    tokens.unshift(CAVAX[chainId] as Token);
+    const _tokens = tokens.filter((token) => token !== CAVAX[chainId]);
+    const currencies = CHAINS[chainId]?.evm ? [CAVAX[chainId], ..._tokens] : _tokens;
+    return currencies.filter((token) => token?.symbol?.toLowerCase() === symbolName.toLowerCase());
+  };
+
+  useEffect(() => {
+    if (defaultInputCurrency) {
+      const filteredInputCurrency = filterCurrencies(defaultInputCurrency);
+      filteredInputCurrency.length > 0 && onCurrencySelection(Field.INPUT, filteredInputCurrency[0]);
+    }
+    if (defaultOutputCurrency && defaultInputCurrency?.toLowerCase() !== inputCurrency?.symbol?.toLowerCase()) {
+      const filteredOutputCurrency = filterCurrencies(defaultOutputCurrency);
+      filteredOutputCurrency.length > 0 && onCurrencySelection(Field.OUTPUT, filteredOutputCurrency[0]);
+    }
+  }, [chainId]);
```

Comment:
this useEffect also should observe `defaultInputCurrency` & `defaultOutputCurrency`